src/wasm/kotlin/java/arcs/wasm/WasmJsInterop.kt

```diff
@@ -13,39 +18,32 @@ import kotlinx.cinterop.toKStringFromUtf8
 import kotlinx.cinterop.toLong
 
 // Model WasmAddress as an Int
-typealias WasmAddress = Int
+typealias WasmAddress = Address
 
 // Wasm Strings are also Int heap pointers
 typealias WasmString = Int
 
 typealias WasmNullableString = Int
 
-/**
- * Any object implementing this interface can be converted into a (pinned) stable heap pointer.
- * To avoid GC Leaks, eventually the ABI should have a Particle.dispose() method which releases
- * these pinned pointers. Right now, the lifetime of these objects depends on the Arcs Runtime
- * holding onto particle and handle references beyond the scope of the call.
- */
-abstract class WasmObject {
-    private var cachedWasmAddress: WasmAddress? = null
-    fun toWasmAddress(): WasmAddress {
-        if (cachedWasmAddress == null) {
-            cachedWasmAddress = StableRef.create(this).asCPointer().toLong().toWasmAddress()
-        }
-        return cachedWasmAddress as WasmAddress
-    }
+fun Any?.toAddress(): Address {
```

Comment:
It looks like this logic is similar to the Addressable file above. Can they be de-duped?

Reply:
I originally tried to have one call the other; I was not successful. @jasonwyatt Do you have any ideas?

Reply:
So, in effect, `Addressable` would be a Wasm-Only implementation (and it wouldn't be publicly exposed)?